### PR TITLE
GraphQL: Merge requisition lists docs into 2.4.2-develop

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) merges the following PRs into 2.4.2-develop.

#8093 
#8354 
#8381 
#8408 

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

whatsnew
Added full GraphQL support for B2B requisition lists. With GraphQL, you can perform the following tasks:

* [Create](https://devdocs.magento.com/guides/v2.4/graphql/mutations/create-requisition-list.html), [update](https://devdocs.magento.com/guides/v2.4/graphql/mutations/update-requisition-list.html), and [delete](https://devdocs.magento.com/guides/v2.4/graphql/mutations/delete-requisition-list.html) requisition lists.
* [Add](https://devdocs.magento.com/guides/v2.4/graphql/mutations/add-products-to-requisition-list.html), [update](https://devdocs.magento.com/guides/v2.4/graphql/mutations/update-requisition-list-items.html), and [delete](https://devdocs.magento.com/guides/v2.4/graphql/mutations/delete-requisition-list-items.html) items in a requisition list.
* [Copy](https://devdocs.magento.com/guides/v2.4/graphql/mutations/copy-items-between-requisition-lists.html) and [move](https://devdocs.magento.com/guides/v2.4/graphql/mutations/move-items-between-requisition-lists.html) items between requisition lists.
* Add requisition list items to the [cart](https://devdocs.magento.com/guides/v2.4/graphql/mutations/add-requisition-list-items-to-cart.html).
* [Clear](https://devdocs.magento.com/guides/v2.4/graphql/mutations/clear-customer-cart.html) the customer cart.
* Return details about requisition lists using the [customer query](https://devdocs.magento.com/guides/v2.4/graphql/queries/customer.html).

### CODE REVIEW

File: src/guides/v2.4/graphql/interfaces/requisition-list-item-interface.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+The `ConfigurableRequisitionListItem` implementation adds the following attributes.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`links`| [DownloadableProductLinks] | An array of links for downloadable products in the requisition list
+`samples` | [DownloadableProductSamples] An array of links to downloadable product samples
+
+### GiftCardRequisitionListItem attributes {#GiftCardRequisitionListItem}
+
+The `GiftCardRequisitionListItem` implementation adds the following attributes.
```
Comment (jeff-matthews) [inline src/guides/v2.4/graphql/interfaces/requisition-list-item-interface.md:58]:
```suggestion
The `GiftCardRequisitionListItem` implementation adds the following attribute.
```

File: src/guides/v2.4/graphql/interfaces/requisition-list-item-interface.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
+---
+group: graphql
+title: RequisitionListItemInterface attributes and implementations
```
Comment (jeff-matthews) [inline src/guides/v2.4/graphql/interfaces/requisition-list-item-interface.md:3]: Does this need the `b2b_only: true` tag? The TOC entry has a B2B tag:https://github.com/magento/devdocs/pull/8428/files#diff-4d2753d92f3de59a53164caf094dc85f939ad66e8b834755573f26ea331d70bbR186

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

File: src/guides/v2.4/graphql/interfaces/requisition-list-item-interface.md (hunk 2 above)
Comment (keharper) [reply src/guides/v2.4/graphql/interfaces/requisition-list-item-interface.md:3]: Good catch. Thanks.

**Review by jeff-matthews — APPROVED** [review]

**Review by cpartica — APPROVED** [review]

File: src/guides/v2.4/graphql/mutations/delete-requisition-list.md [hunk 3]
```diff
@@ -43,7 +51,11 @@ mutation {
 {
   "data": {
     "deleteRequisitionList": {
-      "result": true
+      "status": true,
+      "requisition_lists": {
```
Comment (cpartica) [inline src/guides/v2.4/graphql/mutations/delete-requisition-list.md:55]: i think this needs to return pagination then status

type RequisitionLists @doc(description: "Defines customer requisition lists")  {
    items: [RequisitionList] @doc(description: "An array of requisition lists")
    page_info: SearchResultPageInfo @doc(description: "Contains pagination metadata")
    total_count: Int @doc(description: "The number of returned requisition lists")
}

**Review by cpartica — CHANGES_REQUESTED** [review]

**Review by cpartica — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Pull request merged**